### PR TITLE
Added a way to grant/revoke schema privileges from hive metadata

### PR DESCRIPTION
**Pull request by gaurav8297**

### CODE REVIEW

**Review by kokosing — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 1]
```diff
@@ -2774,6 +2774,18 @@ public void revokeRoles(ConnectorSession session, Set<String> roles, Set<TrinoPr
         return accessControlMetadata.listEnabledRoles(session);
```
Comment (ksobolew) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:2759]: The convention for commit messages is to use imperative mode, like "Add grant/revoke..." and to keep the headline relatively short (I know it can be difficult at times - I struggle with it myself ;) )

(And it's annoying how GitHub does not allow commenting on the commit message)

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/AccessControlMetadata.java [hunk 2]
```diff
@@ -108,20 +108,36 @@ default void revokeRoles(ConnectorSession connectorSession, Set<String> roles, S
...
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support revokes on schemas");
+    }
+
     /**
      * Grants the specified privilege to the specified user on the specified table
      */
     default void grantTablePrivileges(ConnectorSession session, SchemaTableName tableName, Set<Privilege> privileges, HivePrincipal grantee, boolean grantOption)
     {
-        throw new TrinoException(NOT_SUPPORTED, "This connector does not support grants");
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support grants on tables");
```
Comment (Praveen2112) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/AccessControlMetadata.java:132]: nit: I guess this can be captured in a different commit.

**Review by Praveen2112 — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/AccessControlMetadata.java (hunk 2 above)
Comment (kokosing) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/AccessControlMetadata.java:132]: I think this is fine, because without this commit the change is not needed. Doing it sooner or later would be awkward to me.

**Comment by kokosing** [comment]
@cla-bot check

**Comment by kokosing** [comment]
CI hit: https://github.com/trinodb/trino/issues/8432, https://github.com/trinodb/trino/issues/4568
